### PR TITLE
Build: Make TCTIs optional

### PR DESCRIPTION
**Pull request by AndreasFuchsSIT**
Fixes: #917

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -90,6 +90,20 @@ AC_ARG_WITH([tctidefaultconfig],
...
+            [enable_tcti_device=yes])
+AM_CONDITIONAL([ENABLE_TCTI_DEVICE], [test "x$enable_tcti_device" != xno])
+
+AC_ARG_ENABLE([tcti-mssim],
+            [AS_HELP_STRING([--enable-tcti-mssim],
+                            [build the tcti-mssim module (default is yes)])],
+            [enable_tcti_mssim=$enableval],
+            [enable_tcti_mssim=yes])
+AM_CONDITIONAL([ENABLE_TCTI_MSSIM], [test "x$enable_tcti_mssim" != xno])
+
```
Comment (tstruk) [inline configure.ac]: We should check if at least one TCTI is enabled.

Comment (AndreasFuchsSIT) [reply configure.ac]: I intentionally did not do so.

Reason being a case, where someone wants a SAPI in a firmware environment with their own (non-public) TCTI.
Thus, we want to compile without any of our "official" TCTIs.

**Comment by AndreasFuchsSIT** [comment]
@danintel Where and how should these options be mentioned in INSTALL.md ? IMHO INSTALL.md should not be a copy of `./configure --help`. So what do you recommend ?

**Comment by danintel** [comment]
@AndreasFuchsSIT OK--leave it in just` ./configure --help`

File: configure.ac (hunk 1 above)
Comment (AndreasFuchsSIT) [reply configure.ac]: added it to the end, so it is easily spotted.

**Pull request merged**